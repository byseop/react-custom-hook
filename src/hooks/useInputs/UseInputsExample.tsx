import React from 'react';
import useInputs from './useInputs';

function UseInputsExample() {
  const [{ nickname, email }, onChange, reset] = useInputs({
    nickname: '',
    email: '',
  });

  console.log(nickname, email);

  return (
    <>
      <input name="nickname" placeholder="nickname" value={nickname} onChange={e => onChange(e)} />
      <input name="email" placeholder="email" value={email} onChange={e => onChange(e)} />
      <button onClick={reset}>Reset</button>
    </>
  );
}

export default UseInputsExample;
