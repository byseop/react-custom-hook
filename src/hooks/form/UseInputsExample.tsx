import React from 'react';
import useInputs from './useInputs';

function UseInputsExample() {
  const [{ nickname, email }, onChange, reset] = useInputs({
    nickname: '',
    email: '',
  });

  console.log(nickname, email);

  return (
    <>
      <input name="nickname" value={nickname} onChange={e => onChange(e)} />
      <input name="email" value={email} onChange={e => onChange(e)} />
      <button onClick={reset} />
    </>
  );
}

export default UseInputsExample;
