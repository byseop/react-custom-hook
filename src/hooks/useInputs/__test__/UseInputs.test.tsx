import React from 'react';
import UseInputsExample from '../UseInputsExample';
import { render, fireEvent } from '@testing-library/react';

describe('<UseInputsExample />', () => {
  it('matches snapshot', () => {
    const utils = render(<UseInputsExample />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the props correctly', () => {
    const utils = render(<UseInputsExample />);
    utils.getByPlaceholderText('nickname');
    utils.getByPlaceholderText('email');
  })

  it('typing input', () => {
    const utils = render(<UseInputsExample />);
    const nicknameInput = utils.getByPlaceholderText('nickname');
    const emailInput = utils.getByPlaceholderText('email');
    const resetButton = utils.getByText('Reset');

    fireEvent.change(nicknameInput, { target: { value: 'byseop' } });
    fireEvent.change(emailInput, { target: { value: 'byseop@gmail.com' } });

    expect(nicknameInput).toHaveValue('byseop');
    expect(emailInput).toHaveValue('byseop@gmail.com');

    fireEvent.click(resetButton);
    expect(nicknameInput).toHaveValue('');
    expect(emailInput).toHaveValue('');
  });
});
