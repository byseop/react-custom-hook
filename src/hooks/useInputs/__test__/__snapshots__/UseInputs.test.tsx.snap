// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseInputsExample /> matches snapshot 1`] = `
<div>
  <input
    name="nickname"
    placeholder="nickname"
    value=""
  />
  <input
    name="email"
    placeholder="email"
    value=""
  />
  <button>
    Reset
  </button>
</div>
`;
